source/agora/consensus/BlockRewardCalc.d

```diff
@@ -0,0 +1,651 @@
+/*******************************************************************************
+
+    Contains functions to calculate the block reward.
+
+    1. We will only distribute block rewards in every X blocks
+       as opposed to distributing it in every block. This was originally meant
+       to reduce blockchain size.
+    2. We will delay the distribution of the block rewards by Y blocks, so for
+       blocks A .. B we will distribute the rewards at block B + Y.
+       This is because we want to give enough time validators to share
+       block signatures which is the basis of the block reward.
+    3. Each validator will be rewarded based on how many blocks they signed
+       and the reward will NOT be propotional to how much coin they staked.
```

Comment:
I don't quite clearly see why it shouldn't be proportional? Is this explained anywhere?

Reply:
If I remember correctly it was not my decision, but let me know if I am mistaken (@Geod24). A validator who has X * 40k BOA can and will run multiple Agora instances to maximimze its reward, so I agree with you on this.